File: cyber/demo_cc/demo03_client.cc
#include "cyber/cyber.h"
#include "cyber/demo_base_proto/addints.pb.h"

using apollo::cyber::demo_base_proto::AddInts_Request;
using apollo::cyber::demo_base_proto::AddInts_Response;


int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);
    if(argc != 3){
        AWARN << "require two param";
        return 1;
    }
    AINFO << "client ....";
    auto client_node = apollo::cyber::CreateNode("client_node");
    auto client = client_node->CreateClient<AddInts_Request, AddInts_Response>("addints");

    auto req = std::make_shared<AddInts_Request>();
    req->set_num1(std::atoll(argv[1]));
    req->set_num2(std::atoll(argv[2]));

    auto res = client->SendRequest(req);
    AINFO << "res: sum = " << res->sum();

    apollo::cyber::WaitForShutdown();
    return 0;
}

File: cyber/demo_py/demo03_client.py
from cyber.python.cyber_py3 import cyber
from cyber.demo_base_proto.addints_pb2 import AddInts_Request, AddInts_Response
import sys


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("require two param")
        sys.exit(1)

    cyber.init()
    client_node = cyber.Node("client")
    client = client_node.create_client("addints", AddInts_Request, AddInts_Response)

    req = AddInts_Request()
    req.num1 = int(sys.argv[1])
    req.num2 = int(sys.argv[1])

    response = client.send_request(req)
    print("response: sum = {}".format(response.sum))

    cyber.shutdown()